src/ucp/core/ucp_request.h

```diff
@@ -234,11 +234,27 @@ struct ucp_request {
                     /* Remote request ID to acknowledge */
                     ucs_ptr_map_key_t remote_req_id;
 
-                    /* Remote buffer address for get/put operation */
-                    uint64_t          remote_address;
-
-                    /* Key for remote buffer operation */
-                    ucp_rkey_h        rkey;
+                    union {
+                        struct {
+                            /* Remote buffer address for get/put operation */
+                            uint64_t   remote_address;
+                            /* Key for remote buffer operation */
+                            ucp_rkey_h rkey;
+                        };
+                        struct {
+                            /* Remote infomation array, if rma_count != 0 */
+                            struct {
+                                ucp_rkey_h rkey;
+                                uint64_t   remote_address;
+                                size_t     remote_size;
+                                size_t     accumulate_size;
+                            } * rma_array;
+                            /* Index of rma_array, if if rma_count != 0 */
+                            size_t rma_index;
+                        };
+                    };
+                    /* Key counts for remote buffer operation */
+                    size_t rma_count;
```

Comment:
done

Reply:
I would also propose to rename `remote_address` and `remote_size` to `address` and `size` accordingly. WDYTA?

Reply:
done

Reply:
Sorry, misunderstanding happened.

I meant to change `remote_address` and `remote_size` to `address` and `size` without `r` prefix since they are already inside `rdata` member that already contains this `r` part.

Reply:
done